addons/main.go

```diff
@@ -107,6 +111,14 @@ func parseAddonFlags(addonFlags *addonFlags) {
 	flag.BoolVar(&addonFlags.enableLeaderElection, "leader-elect", false,
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.")
+	flag.StringVar(&addonFlags.leaderElectionID, "leader-elect-id", constants.AddonControllerManagerLeaderElectionResourceName,
+		"The name of the resource that leader election will use for holding the leader lock.")
+	flag.DurationVar(&addonFlags.leaderElectionLeaseDuration, "leader-elect-lease-duration", 15*time.Second,
+		"Interval at which non-leader candidates will wait to force acquire leadership")
+	flag.DurationVar(&addonFlags.leaderElectionRenewDeadline, "leader-elect-renew-deadline", 10*time.Second,
+		"Duration that the leading controller manager will retry refreshing leadership before giving up")
+	flag.DurationVar(&addonFlags.leaderElectionRetryPeriod, "leader-elect-retry-period", 2*time.Second,
```

Comment:
Question: Which code uses these flags?

Reply:
These flags are used in the initialization of manager `Options`.